conda/models/channel.py

```diff
@@ -368,7 +368,7 @@ def _get_channel_for_name_helper(name):
         # stripping off path threw information away from channel_name (i.e. any potential subname)
         # channel.name *should still be* channel_name
         channel = copy(channel)
-        channel.name = channel_name
+        channel.name = _stripped
```

Comment:
This one-line fix could be a separate PR. But oh well. It's at least in its own commit.  The commit has a test, which is included in `tests/models/test_channel.py`.